scala-mode2.el

```diff
@@ -169,5 +170,11 @@ When started, runs `scala-mode-hook'.
                '("\\.\\(scala\\|sbt\\)\\'" . scala-mode))
   (modify-coding-system-alist 'file "\\.\\(scala\\|sbt\\)\\'" 'utf-8))
 
+(defvar scala--prettify-symbols-alist scala-mode-pretty-all-alist)
+(defun scala-add-pretty ()
+  (setq prettify-symbols-alist scala--prettify-symbols-alist))
+
+(add-hook 'scala-mode-hook 'scala-add-pretty)
```

Comment:
I'm not sure we should do this by default, because prettify was only added to Emacs relatively recently.

What I propose is that you keep the default setting to enable prettifications for all symbols (create a `defcustom` to replace the current `-all` constant) but then we document that people need to add the hook to get the support... and then they can go modify the `alist` if they want to restrict or add to the default list.